docs/_docs/plugins/generators.md

```diff
@@ -3,78 +3,112 @@ title: Generators
+* The pages are intended to render a list of all documents under a given category. So the basename of the rendered file
+would be better as `index.html`.
+* Having the ability to configure the pages via [front matter defaults](/docs/configuration/front-matter-defaults/)
+would be awesome! So assigning a particular `type` to these pages would be beneficial.
 
 ```ruby
-module Jekyll
-  class CategoryPageGenerator < Generator
+module SamplePlugin
+  class CategoryPageGenerator < Jekyll::Generator
     safe true
 
     def generate(site)
-      if site.layouts.key? 'category_index'
-        dir = site.config['category_dir'] || 'categories'
-        site.categories.each_key do |category|
-          site.pages << CategoryPage.new(site, site.source, File.join(dir, category), category)
-        end
+      site.categories.each do |category, posts|
+        site.pages << CategoryPage.new(site, category, posts)
       end
     end
   end
 
-  # A Page subclass used in the `CategoryPageGenerator`
-  class CategoryPage < Page
-    def initialize(site, base, dir, category)
-      @site = site
-      @base = base
-      @dir  = dir
-      @name = 'index.html'
-
-      self.process(@name)
-      self.read_yaml(File.join(base, '_layouts'), 'category_index.html')
-      self.data['category'] = category
+  # Subclass of `Jekyll::Page` with custom method definitions.
+  class CategoryPage < Jekyll::Page
+    def initialize(site, category, posts)
+      @site = site             # the current site instance.
+      @base = site.source      # path to the source directory.
+      @dir  = category         # the directory the page will reside in.
+
+      # All pages have the same filename, so define attributes straight away.
+      @basename = 'index'      # filename without the extension.
+      @ext      = '.html'      # the extension.
+      @name     = 'index.html' # basically @basename + @ext.
+
+      # Initialize data hash with a key pointing to all posts under current category.
+      # This allows accessing the list in a template via `page.linked_docs`.
+      @data = {
+        'linked_docs' => posts
+      }
+
+      # Look up front matter defaults scoped to type `categories`, if given key
+      # doesn't exist in the `data` hash.
+      data.default_proc = proc do |_, key|
+        site.frontmatter_defaults.find(relative_path, :categories, key)
+      end
+    end
 
-      category_title_prefix = site.config['category_title_prefix'] || 'Category: '
-      self.data['title'] = "#{category_title_prefix}#{category}"
+    # Placeholders that are used in constructing page URL.
+    def url_placeholders
+      {
+        :category   => @dir,
+        :basename   => basename,
+        :output_ext => output_ext,
+      }
     end
   end
 end
 ```
+The generated pages can now be set up to use a particular layout or output at a particular path in the destination
```

Comment:
```suggestion

The generated pages can now be set up to use a particular layout or output at a particular path in the
```

Reply:
Why? The lines have been normalized to be 120 chars or less..

Reply:
Sorry was missing a word. Updated now.

My suggestion was an empty line before the line.